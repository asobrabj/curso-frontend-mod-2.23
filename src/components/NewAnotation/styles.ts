import styled from 'styled-components'

export const NewAnotation = styled.div`
  position: absolute;
  top: 10px;
  right: 70px;
  display: flex;
  align-items: center;
  cursor: pointer;
  padding: 8px;
`
export const Paragraph = styled.p`
  font-weight: bold;
  margin-right: 8px;
`

export const Icon = styled.i`
  font-size: 24px;
  margin-top: 5px;
`
