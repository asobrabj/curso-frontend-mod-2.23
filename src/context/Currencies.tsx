import { createContext, useContext, useState } from 'react';
import instance from '../service/axiosaInstace';

export interface ICurrencyContext {
  to: string;
  setTo: React.Dispatch<React.SetStateAction<string>>;
  from: string;
  setFrom: React.Dispatch<React.SetStateAction<string>>;
  amount: string;
  setAmount: React.Dispatch<React.SetStateAction<string>>;
  fecthConversion: (
    to: string,
    from: string,
    amount: string,
  ) => Promise<string>;
  error: boolean;
}

interface ICurrencyProvider {
  children: React.ReactNode;
}

const CurrencyContext = createContext<ICurrencyContext>({} as ICurrencyContext);

export const CurrencyProvider: React.FC<ICurrencyProvider> = ({ children }) => {
  const [error, setError] = useState<boolean>(false);
  const [to, setTo] = useState<string>('');
  const [from, setFrom] = useState<string>('');
  const [amount, setAmount] = useState<string>('');

  const fecthConversion = async (
    to: string,
    from: string,
    amount: string,
  ): Promise<string> => {
    try {
      const response = await instance.get(`${to}/${from}/${amount}`);
      return response.data.conversion_result;
    } catch (err) {
      setError(true);
      return 'Erro na convers√£o';
    }
  };

  return (
    <CurrencyContext.Provider
      value={{
        fecthConversion,
        error,
        amount,
        from,
        to,
        setTo,
        setAmount,
        setFrom,
      }}
    >
      {children}
    </CurrencyContext.Provider>
  );
};

export const useCurrency = () => {
  return useContext(CurrencyContext);
};
