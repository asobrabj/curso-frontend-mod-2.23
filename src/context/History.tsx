/* eslint-disable react-refresh/only-export-components */
import { createContext, useContext, useEffect, useState } from 'react';

export interface IDataCurrency {
  to: string;
  from: string;
  amount: string;
  result: string;
  date: string;
}

export interface IHistoryContext {
  setHistory: (data: IDataCurrency) => void;
  history?: Array<IDataCurrency>;
}

interface IHistoryProvider {
  children: React.ReactNode;
}

const HistoryContext = createContext<IHistoryContext>({} as IHistoryContext);

export const HistoryProvider: React.FC<IHistoryProvider> = ({ children }) => {
  const [history, setHistoryState] = useState<Array<IDataCurrency>>();

  useEffect(() => {
    const currentData = window.localStorage.getItem('history');
    const newData = currentData ? JSON.parse(currentData) : [];
    setHistoryState(newData);
  }, []);

  const setHistory = (data: IDataCurrency) => {
    const currentData = window.localStorage.getItem('history');
    const newData = currentData ? JSON.parse(currentData) : [];
    newData.push(data);
    setHistoryState(newData);
    window.localStorage.setItem('history', JSON.stringify(newData));
    return;
  };

  return (
    <HistoryContext.Provider value={{ setHistory, history }}>
      {children}
    </HistoryContext.Provider>
  );
};

export const useHistory = () => {
  return useContext(HistoryContext);
};
