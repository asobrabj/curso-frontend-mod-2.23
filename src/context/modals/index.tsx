import { createContext, useContext, useState } from 'react'
import IAnotation from '../../types/anotation'

export interface IModalContext {
  isVisible: boolean
  setIsVisible: React.Dispatch<React.SetStateAction<boolean>>
  isForm: boolean
  setIsForm: React.Dispatch<React.SetStateAction<boolean>>
  dataCard: IAnotation
  setDataCard: React.Dispatch<React.SetStateAction<IAnotation>>
  idCard: number | null
  setIdCard: React.Dispatch<React.SetStateAction<number | null>>
}

interface IModalProvider {
  children: React.ReactNode
}

const ModalContext = createContext<IModalContext>({} as IModalContext)

export function ModalProvider({ children }: IModalProvider) {
  const [isVisible, setIsVisible] = useState<boolean>(false)
  const [isForm, setIsForm] = useState<boolean>(false)
  const [idCard, setIdCard] = useState<number | null>(null)
  const [dataCard, setDataCard] = useState<IAnotation>({
    id: null,
    concluded: false,
    date: '',
    description: '',
    name: '',
    category: '',
    priority: '',
    time: '',
  })

  return (
    <ModalContext.Provider
      value={{
        idCard,
        setIdCard,
        isVisible,
        setIsVisible,
        isForm,
        setIsForm,
        dataCard,
        setDataCard,
      }}
    >
      {children}
    </ModalContext.Provider>
  )
}

export function useModal() {
  return useContext(ModalContext)
}
