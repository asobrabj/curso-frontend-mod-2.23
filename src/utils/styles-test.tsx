import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from '../styles/darkTheme'

export const renderTheme = (Children: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}> {Children} </ThemeProvider>)
}
