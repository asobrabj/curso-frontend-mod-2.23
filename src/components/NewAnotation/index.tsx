import { CiCirclePlus } from 'react-icons/ci'
import { useModal } from '../../context/modals'
import * as S from './styles'

export default function NewAnotation() {
  const { setIsVisible, setIsForm } = useModal()

  const handleNewAnotation = () => {
    setIsForm(true)
    setIsVisible(true)
  }
  return (
    <S.NewAnotation onClick={handleNewAnotation} title="new-anotation">
      <S.Paragraph>Nova Anotação</S.Paragraph>
      <S.Icon>
        <CiCirclePlus />
      </S.Icon>
    </S.NewAnotation>
  )
}
