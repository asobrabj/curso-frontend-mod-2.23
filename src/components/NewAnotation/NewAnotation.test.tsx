import { act, fireEvent, waitFor } from '@testing-library/react'
import NewAnotation from '.'
import { useModal } from '../../context/modals'
import { renderTheme } from '../../utils/styles-test'

jest.mock('../../context/modals', () => ({
  ...jest.requireActual('../../context/modals'), // Preserva as implementações não-mockadas
  useModal: () => ({
    setIsVisible: jest.fn(),
    setIsForm: jest.fn(),
  }),
}))

describe('NewAnotation Component', () => {
  test('Renderiza o componente corretamente', () => {
    const { getByText, getByTitle } = renderTheme(<NewAnotation />)

    expect(getByText('Nova Anotação')).toBeInTheDocument()
    expect(getByTitle('new-anotation')).toBeInTheDocument()
  })

  test('Chama as funções setIsForm e setIsVisible quando clicado', () => {
    const { getByTitle } = renderTheme(<NewAnotation />)
    act(() => {
      fireEvent.click(getByTitle('new-anotation'))
    })
    waitFor(() => {
      expect(useModal().setIsForm).toHaveBeenCalledWith(true)
      expect(useModal().setIsVisible).toHaveBeenCalledWith(true)
    })
  })
  it('snapshot', () => {
    const { container } = renderTheme(<NewAnotation />)
    expect(container).toMatchSnapshot()
  })
})
