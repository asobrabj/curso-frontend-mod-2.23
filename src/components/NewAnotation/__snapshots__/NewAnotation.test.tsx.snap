// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewAnotation Component snapshot 1`] = `
.c0 {
  position: absolute;
  top: 10px;
  right: 70px;
  display: flex;
  align-items: center;
  cursor: pointer;
  padding: 8px;
}

.c1 {
  font-weight: bold;
  margin-right: 8px;
}

.c2 {
  font-size: 24px;
  margin-top: 5px;
}

<div>
   
  <div
    class="c0"
    title="new-anotation"
  >
    <p
      class="c1"
    >
      Nova Anotação
    </p>
    <i
      class="c2"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="Circle_Plus"
        >
          <g>
            <path
              d="M15,12.5H12.5V15a.5.5,0,0,1-1,0V12.5H9a.5.5,0,0,1,0-1h2.5V9a.5.5,0,0,1,1,0v2.5H15A.5.5,0,0,1,15,12.5Z"
            />
            <path
              d="M12,21.932A9.934,9.934,0,1,1,21.932,12,9.944,9.944,0,0,1,12,21.932ZM12,3.065A8.934,8.934,0,1,0,20.932,12,8.944,8.944,0,0,0,12,3.065Z"
            />
          </g>
        </g>
      </svg>
    </i>
  </div>
   
</div>
`;
